File: user/src/main/kotlin/kpring/user/exception/ErrorCode.java
package kpring.user.exception;

import org.springframework.http.HttpStatus;

public enum ErrorCode {
    // BAD_REQUEST
    ALREADY_EXISTS_EMAIL(HttpStatus.BAD_REQUEST, 4001, "Email already exists"),
    INVALID_EMAIL(HttpStatus.BAD_REQUEST, 4002, "Invalid email"),
    ;
    public final HttpStatus status;
    public final int code;
    public final String message;

    ErrorCode(HttpStatus status, int code, String message) {
        this.status = status;
        this.code = code;
        this.message = message;
    }

    public boolean isServerError(){
        return status.is5xxServerError();
    }
}

File: chat/src/main/kotlin/kpring/chat/exception/ErrorCode.kt
package kpring.chat.exception

import org.springframework.http.HttpStatus

enum class ErrorCode(val httpStatus: Int, val message: String) {

    //400
    INVALID_TOKEN_BODY(HttpStatus.NOT_FOUND.value(), "토큰의 body가 유효하지 않습니다"),

}